import "building.proto";
import "hero.proto";
import "node.proto";
import "item.proto";
import "resource.proto";
import "mail.proto";

package SanProto;

//资源开发
message ExploitReq {
  required uint64 user_id = 1;
  required NodeInfo node = 2;
  repeated HeroInfo heroes = 3;       //参与采集的英雄
  repeated BuildingInfo buildings = 4;//相关联的建筑物（会从其中抽调英雄）
}
message ExploitRes {
  enum RET {
    OK = 0;
    NOT_EXPLOITING = 1;
    CANNT_FINISH = 2;
  }
  required int32 status = 1;
  optional ResourceInfo resource = 2;
  repeated ItemInfo items = 3;
  optional MailInfo mail = 4;
  optional RET ret = 5;
  optional NodeInfo node = 6;
}

//资源采集
message GatherReq {
  required uint64 user_id = 1;
  repeated NodeInfo nodes = 2;
}
message GatherRes {
  required int32 status = 1;
  optional ResourceInfo resource = 2;
  repeated ItemInfo items = 3;
}

